runtime/doc/options.txt

```diff
@@ -1327,7 +1327,10 @@ A jump table for the options with a short description can be found at |Q_op|.
 	Number of screen lines to use for the command-line.  Helps avoiding
 	|hit-enter| prompts.
 	The value of this option is stored with the tab page, so that each tab
-	page can have a different value.
+	page can have a different value. Setting it to '0' hides cmdline and is
+  appropriate for windows that user is not supposed to use cmdline in.
+  If you use cmdline while 'cmdheight' is set to '0' it'll behave
+  strangely specially when it interacts with statusline.
```

Comment:
It does its work which is "hiding cmdline for windows in which user is not supposed to use cmdline" there are many use cases for it. ctrlp has lots of code just to simulate this behavior (which could be done by changing one bit in vim source) my fzf looks much better with cmdline=0 for its window. I understand it may introduce bug reports so the tradeoff is obvious:
1. merge this patch and let plugin developers hide cmdline in their plugin's window, and explain the behavior in first bug report and ask people interested in zero height cmdline for their normal vim usage to provide pr and reference other bug reports to that first bug report.
2. don't merge this patch to reduce bug reports and the annoyance.